integration-tests/inc/options/test-class-wpseo-option-social.php

```diff
@@ -72,6 +73,49 @@ public function validate_option_provider() {
 				'clean'    => array( 'og_frontpage_image_id' => '' ),
 				'old'      => array(),
 			),
+			array(
+				'expected' => array( 'facebook_site' => '' ),
+				'dirty'    => array( 'facebook_site' => 'invalidurl' ),
+				'clean'    => array( 'facebook_site' => '' ),
+				'old'      => array(),
+			),
+			array(
+				'expected' => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'dirty'    => array( 'youtube_url' => 'invalidurl' ),
+				'clean'    => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'old'      => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+			),
+			array(
+				'expected' => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'dirty'    => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'clean'    => array( 'youtube_url' => 'https://www.youtube.com/yoast' ),
+				'old'      => array(),
+			),
 		);
 	}
+
+	/**
+	 * Tests that submitting an option with an invalid URL adds a WordPress settings error notice.
+	 *
+	 * @covers WPSEO_Option_Social::validate_url
+	 */
+	public function test_validate_url_adds_settings_error() {
```

Comment:
I think a better way to test this is to check if add_settings_error is called once (with the arguments you expect). We don't want to use WordPress globals in tests. When we're moving it to a brain monkey test, that wouldn't even be possible.